File: src/main/java/club/deneb/client/client/NotificationManager.java
package club.deneb.client.client;

import club.deneb.client.features.AbstractModule;
import club.deneb.client.utils.Timer;
import java.util.List;
import java.util.concurrent.CopyOnWriteArrayList;

/**
 * Author B_312 on 01/15/2021
 */
public class NotificationManager {

    public static NotificationManager INSTANCE;

    private final List<NotificationUnit> notificationUnitsList = new CopyOnWriteArrayList<>();

    public static class NotificationUnit{
        public Timer timer;
        public AbstractModule module;
        public boolean enabled;

        public NotificationUnit(Timer timer, AbstractModule module, boolean enabled){
            this.module = module;
            this.timer = timer;
            this.enabled = enabled;
        }
    }

    public NotificationManager(){
        INSTANCE = this;
    }

    public static void addNewNotification(AbstractModule module, boolean enabled){
        Timer tempTimer = new Timer();
        tempTimer.reset();
        INSTANCE.notificationUnitsList.add(new NotificationUnit(tempTimer,module,enabled));
    }

    public static List<NotificationUnit> getUnits(){
        return INSTANCE.notificationUnitsList;
    }



}

File: src/main/java/club/deneb/client/client/NotificationManager.kt
package club.deneb.client.client

import java.util.concurrent.CopyOnWriteArrayList
import club.deneb.client.features.AbstractModule
import club.deneb.client.utils.Timer

/**
 * Author B_312 on 01/15/2021
 */
object NotificationManager {

    val notificationUnitsList: MutableList<NotificationUnit> = CopyOnWriteArrayList()

    class NotificationUnit(var timer: Timer, var module: AbstractModule, var enabled: Boolean)

    @JvmStatic
    fun addNewNotification(module: AbstractModule, enabled: Boolean) {
        val tempTimer = Timer()
        tempTimer.reset()
        notificationUnitsList.add(NotificationUnit(tempTimer, module, enabled))
    }

}